### PR TITLE
Storybook 7 migration

### LINKED ISSUE #900: 4.x - Update Storybook from 6 to 7
Update Storybook from 6 to 7

# Proposed Solution

- [x] Upgrade from 6 to 7 Storybook https://storybook.js.org/docs/vue/migration-guide
- [ ] Look at the log messages and clean up anything not used
- [ ] Figure out how we want the docs to render for each component, currently I have set up a separate "docs" page for each component where we can outline "how to use" where we could provide CSS and HTML source blocks.  View the card component `CardDocs.mdx` file as an example. 
- [x] Figure out how to render HTML as the old HTML tab is not working I think
- [ ] `https://github.com/storybookjs/storybook/issues/13917` might be an issue we are having
- [ ] Possibly figure out how to how to provide a `component-standalone.css` file that contains, fonts, base.css, and component.css - see https://codepen.io/bspeare/pen/poGOjzm as an example. 
- [x] "Not implemented" docs have extra stuff that can be removed/hidden

### PR DESCRIPTION
**Pull request by bspeare**
Resolves https://github.com/uiowa/uids/issues/900

Don't have NVM? https://github.com/nvm-sh/nvm?tab=readme-ov-file#git-install

`nvm install 18`
`nvm use`
`yarn -v` _should output 3.7.0 otherwise you may need to run `corepack enable` while on node 18._
`yarn install`
`yarn storybook`

- Should only have one warning on startup: `WARN You (or an addon) are using the 'config' preset field. This has been replaced by 'previewAnnotations' and will be removed in 8.0`
- Introduction docs load
- Each story should have `no code available`
- Each not implemented story should just be the story without the controls and other doc-isms.
- HTML tab should show on each implemented component.
- See passing test build in GH Actions
- BONUS: See uiowa favicon

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,43 +23,52 @@
...
-    "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "storybook": "storybook dev -p 6006",
+    "build-storybook": "storybook build"
   },
   "dependencies": {
     "vue": "^3.2.31",
     "yarn": "^1.22.19"
   },
+  "resolutions": {
```
Comment (bspeare) [inline package.json]: Added this based on `https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092`

**Comment by joewhitsitt** [comment]
i'll come back to this tomorrow.

**Comment by joewhitsitt** [comment]
Heads up, I am fine if we decide to pull out the yarn 3 stuff for now. I was just happy to get it to work. Realized along the way how different it is.

**Comment by bspeare** [comment]
I am noticing that our HTML now contains the slot comments.  Can we remove these comments for figure out a comment syntax that is not rendered with the component?

Screenshot 2023-12-13 at 11 28 16 AM

**Comment by bspeare** [comment]
Now that we have a dedicated docs page, can we hide the story page for each component that is not implemented?

Screenshot 2023-12-13 at 11 34 47 AM

**Comment by joewhitsitt** [comment]
Good questions. However, I am now running into this locally. Not sure what changed...
>Failed to fetch dynamically imported module: http://localhost:6006/.storybook/preview.js

Update: had to hard cache reload. but did a bunch of things before that so idk.

**Comment by joewhitsitt** [comment]
>Now that we have a dedicated docs page, can we hide the story page for each component that is not implemented?

>figure out a comment syntax that is not rendered with the component?

Sounds cool. Looking into to both of these, but haven't had much luck.

**Comment by joewhitsitt** [comment]
Can hide the whole expandable sidebar item using:

in .storybook/manager.js
```
  sidebar: {
    filters: {
      patterns: (item) => {
        return !item.tags.includes('disabled');
      }
    }
  }
```
and in each story:

```
export default {
  title: 'Not implemented/Background',
  // https://github.com/storybookjs/storybook/issues/14442#issuecomment-1089165153
  tags: ['disabled'],
```

But this is not what we want. If anything it seems like we want to prevent docs and just show the story.

**Comment by bspeare** [comment]
@joewhitsitt could you post a quick screenshot of what that would look like?

**Comment by joewhitsitt** [comment]
The routes still exist but it is not in the sidebar.

http://localhost:6006/?path=/story/not-implemented-background--background
http://localhost:6006/?path=/docs/not-implemented-background--docs

Screenshot 2023-12-14 at 10 09 06 AM

**Comment by joewhitsitt** [comment]
We could create other doc pages like `Introduction` but probably can't leverage the current story Stub logic to construct the docs.

**Review by bspeare — COMMENTED** [review]
Tested and I would approve if I could.  @pyrello do you want to review?

**Comment by joewhitsitt** [comment]
Created https://github.com/uiowa/uids/issues/912 and https://github.com/uiowa/uids/issues/913 to capture Ben's nice-to-haves

**Comment by pyrello** [comment]
Pulling to review

**Review by pyrello — APPROVED** [review]
Changes look good.

**Pull request merged**